docs/learn/response_models.md

```diff
@@ -24,9 +24,9 @@ class Book(BaseModel):
 
 
 @openai.call(model="gpt-4o-mini", response_model=Book)
-@prompt_template("Recommend a {genre} book")
-def recommend_book(genre: str):
-    ...
+@prompt_template()
+def recommend_book(genre: str) -> str:
```

Comment:
nit: should return `Messages.Type`?

Reply:
There is nothing wrong with setting this to `Messages.Type`, but I have one question.
If I were a user, I would want to define it as str because it would be a hassle to type `Message.Type`.
Typing-wise, Message.Type is fine because it encompasses `str`.
Are you saying that you are writing `Message.Type` as a reference here? Or are you strongly suggesting that we deprecate writing str and force users to use `Message.Type`?

Reply:
hmm, i guess it's fine to type it as str, i was mostly suggesting wrt. consistency of showing returning `Messages.Type` for easy extension down the line, but i guess typing it as str does remove the need for the extra import unless you're using it